htp/htp_decompressors.c

```diff
@@ -295,12 +295,18 @@ static htp_status_t htp_gzip_decompressor_decompress(htp_decompressor_gzip_t *dr
                 drec->stream.next_in += inprocessed;
                 drec->stream.avail_out -= outprocessed;
                 drec->stream.next_out += outprocessed;
-                if (status == LZMA_STATUS_FINISHED_WITH_MARK) {
-                    rc = Z_STREAM_END;
-                } else if (rc == SZ_OK) {
-                    rc = Z_OK;
-                } else {
-                    rc = Z_DATA_ERROR;
+                switch (status) {
+                    case LZMA_STATUS_FINISHED_WITH_MARK:
+                        rc = Z_STREAM_END;
+                        break;
+                    case SZ_OK:
+                        rc = Z_OK;
+                        break;
+                    case SZ_ERROR_MEM:
+                        htp_log(d->tx->connp, HTP_LOG_MARK, HTP_LOG_WARNING, 0, "LZMA decompressor: memory limit reached");
+                        // no break
+                    default:
+                        rc = Z_DATA_ERROR;
```

Comment:
can we get a more specific error to log here? I think Z_DATA_ERROR gets us a generic error.